# -*- coding: utf-8 -*-


# Verändern falls neue .ui Datei:
#   1. Pfad für icons erstellen :
#         cwd = Path.cwd()
#         rootdir = cwd.parent.absolute()
#         iconsdir = rootdir / "icons"
#   2. Pfade anpassen mit (str(iconsdir/"Bildname.PNG"))
#   3. Getter einfügen
#----------------------------------------------------------
# Neue View in Projekt einpflegen:
#   1. form.ui mit designer erstellen
#   2. pyuic5 -x form.ui -o form.py



from PyQt5 import QtCore, QtGui, QtWidgets
from pathlib import Path
from sys import platform


class Ui_View(object):
    def setupUi(self, View):
        cwd = Path.cwd()
        rootdir = cwd.parent.absolute()

        if platform == "linux" or platform == "win32":
            iconsdir = rootdir / "icons_win"
        else:
            iconsdir = rootdir / "icons_mac"

        View.setObjectName("View")
        View.resize(1232, 768)
        self.centralwidget = QtWidgets.QWidget(View)
        self.centralwidget.setObjectName("centralwidget")
        self.graphicsView = QtWidgets.QLabel(self.centralwidget)
        self.graphicsView.setGeometry(QtCore.QRect(0, 0, 1231, 771))
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Maximum, QtWidgets.QSizePolicy.Maximum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.graphicsView.sizePolicy().hasHeightForWidth())
        self.graphicsView.setSizePolicy(sizePolicy)
        self.graphicsView.setLayoutDirection(QtCore.Qt.RightToLeft)
        self.graphicsView.setLineWidth(0)
        self.graphicsView.setText("")
        self.graphicsView.setScaledContents(True)
        self.graphicsView.setAlignment(QtCore.Qt.AlignLeading | QtCore.Qt.AlignLeft | QtCore.Qt.AlignTop)
        self.graphicsView.setIndent(0)
        self.graphicsView.setObjectName("graphicsView")
        View.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(View)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 1232, 22))
        self.menubar.setObjectName("menubar")
        self.menubar.setNativeMenuBar(False)
        self.menuFile = QtWidgets.QMenu(self.menubar)
        self.menuFile.setObjectName("menuFile")
        self.menuPaint_Settings = QtWidgets.QMenu(self.menubar)
        self.menuPaint_Settings.setObjectName("menuPaint_Settings")
        View.setMenuBar(self.menubar)
        self.toolBar = QtWidgets.QToolBar(View)
        self.toolBar.setObjectName("toolBar")
        View.addToolBar(QtCore.Qt.TopToolBarArea, self.toolBar)
        self.toolBar.setMaximumSize(1232, 25)
        self.actionPaint = QtWidgets.QAction(View)
        self.actionPaint.setCheckable(True)
        icon9 = QtGui.QIcon()
        icon9.addPixmap(QtGui.QPixmap(str(iconsdir/"zeichnen.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionPaint.setIcon(icon9)
        self.actionPaint.setObjectName("actionPaint")
        self.actionErase = QtWidgets.QAction(View)
        self.actionErase.setCheckable(True)
        icon10 = QtGui.QIcon()
        icon10.addPixmap(QtGui.QPixmap(str(iconsdir/"radierer.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionErase.setIcon(icon10)
        self.actionErase.setObjectName("actionErase")
        self.actionThick = QtWidgets.QAction(View)
        self.actionThick.setCheckable(True)
        icon11 = QtGui.QIcon()
        icon11.addPixmap(QtGui.QPixmap(str(iconsdir/"dick.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionThick.setIcon(icon11)
        self.actionThick.setObjectName("actionThick")
        self.actionMedium = QtWidgets.QAction(View)
        self.actionMedium.setCheckable(True)
        icon12 = QtGui.QIcon()
        icon12.addPixmap(QtGui.QPixmap(str(iconsdir/"mittel.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionMedium.setIcon(icon12)
        self.actionMedium.setObjectName("actionMedium")
        self.actionThin = QtWidgets.QAction(View)
        self.actionThin.setCheckable(True)
        icon13 = QtGui.QIcon()
        icon13.addPixmap(QtGui.QPixmap(str(iconsdir/"duenn.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionThin.setIcon(icon13)
        self.actionThin.setObjectName("actionThin")
        self.actionRed = QtWidgets.QAction(View)
        self.actionRed.setCheckable(True)
        icon14 = QtGui.QIcon()
        icon14.addPixmap(QtGui.QPixmap(str(iconsdir/"red.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionRed.setIcon(icon14)
        self.actionRed.setObjectName("actionRed")
        self.actionBlue = QtWidgets.QAction(View)
        self.actionBlue.setCheckable(True)
        icon15 = QtGui.QIcon()
        icon15.addPixmap(QtGui.QPixmap(str(iconsdir/"blue.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionBlue.setIcon(icon15)
        self.actionBlue.setObjectName("actionBlue")
        self.actionYellow = QtWidgets.QAction(View)
        self.actionYellow.setCheckable(True)
        icon16 = QtGui.QIcon()
        icon16.addPixmap(QtGui.QPixmap(str(iconsdir/"yellow.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionYellow.setIcon(icon16)
        self.actionYellow.setObjectName("actionYellow")
        self.actionSpacer1 = QtWidgets.QAction(View)
        self.actionSpacer1.setEnabled(False)
        self.actionSpacer2 = QtWidgets.QAction(View)
        self.actionSpacer2.setEnabled(False)
        self.actionSpacer3 = QtWidgets.QAction(View)
        self.actionSpacer3.setEnabled(False)
        self.actionSpacer4 = QtWidgets.QAction(View)
        self.actionSpacer4.setEnabled(False)
        self.actionSpacer5 = QtWidgets.QAction(View)
        self.actionSpacer5.setEnabled(False)
        self.actionSpacer6 = QtWidgets.QAction(View)
        self.actionSpacer6.setEnabled(False)
        self.actionSpacer7 = QtWidgets.QAction(View)
        self.actionSpacer7.setEnabled(False)
        self.actionUndo = QtWidgets.QAction(View)
        icon17 = QtGui.QIcon()
        icon17.addPixmap(QtGui.QPixmap(str(iconsdir/"undo.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionUndo.setIcon(icon17)
        self.actionUndo.setObjectName("actionUndo")
        self.actionRedo = QtWidgets.QAction(View)
        icon18 = QtGui.QIcon()
        icon18.addPixmap(QtGui.QPixmap(str(iconsdir/"redo.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionRedo.setIcon(icon18)
        self.actionRedo.setObjectName("actionRedo")
        self.actionClear_all = QtWidgets.QAction(View)
        self.actionClear_all.setObjectName("actionClear_all")
        self.actionOpen = QtWidgets.QAction(View)
        self.actionOpen.setObjectName("actionOpen")
        self.actionSave = QtWidgets.QAction(View)
        self.actionSave.setObjectName("actionSave")
        self.actionCalibrate = QtWidgets.QAction(View)
        self.actionCalibrate.setObjectName("actionCalibrate")
        self.actionHelp = QtWidgets.QAction(View)
        self.actionHelp.setObjectName("actionHelp")
        self.actionColorpicker = QtWidgets.QAction(View)
        icon19 = QtGui.QIcon()
        icon19.addPixmap(QtGui.QPixmap(str(iconsdir/"Farbe.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionColorpicker.setIcon(icon19)
        self.actionColorpicker.setObjectName("actionColorpicker")
        self.actionStrokewidth = QtWidgets.QAction(View)
        icon20 = QtGui.QIcon()
        icon20.addPixmap(QtGui.QPixmap(str(iconsdir/"Strichdicke.PNG")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        self.actionStrokewidth.setIcon(icon20)
        self.actionStrokewidth.setObjectName("actionStrokewidth")
        self.menuFile.addAction(self.actionClear_all)
        self.menuFile.addAction(self.actionOpen)
        self.menuFile.addAction(self.actionSave)
        self.menuFile.addAction(self.actionCalibrate)
        self.menuFile.addAction(self.actionHelp)
        self.menuPaint_Settings.addAction(self.actionColorpicker)
        self.menuPaint_Settings.addAction(self.actionStrokewidth)
        self.menubar.addAction(self.menuFile.menuAction())
        self.menubar.addAction(self.menuPaint_Settings.menuAction())
        self.toolBar.addAction(self.actionPaint)
        self.toolBar.addAction(self.actionErase)
        self.toolBar.addAction(self.actionSpacer1)
        self.toolBar.addAction(self.actionRed)
        self.toolBar.addAction(self.actionBlue)
        self.toolBar.addAction(self.actionYellow)
        self.toolBar.addAction(self.actionSpacer2)
        self.toolBar.addAction(self.actionThick)
        self.toolBar.addAction(self.actionMedium)
        self.toolBar.addAction(self.actionThin)
        self.toolBar.addAction(self.actionSpacer3)
        self.toolBar.addAction(self.actionSpacer4)
        self.toolBar.addAction(self.actionSpacer5)
        self.toolBar.addAction(self.actionSpacer6)
        self.toolBar.addAction(self.actionSpacer7)
        self.toolBar.addAction(self.actionUndo)
        self.toolBar.addAction(self.actionRedo)

        self.retranslateUi(View)
        QtCore.QMetaObject.connectSlotsByName(View)

    def retranslateUi(self, View):
        _translate = QtCore.QCoreApplication.translate
        View.setWindowTitle(_translate("View", "View"))
        self.menuFile.setTitle(_translate("View", "File"))
        self.menuPaint_Settings.setTitle(_translate("View", "Paint Settings"))
        self.toolBar.setWindowTitle(_translate("View", "toolBar"))
        self.actionPaint.setText(_translate("View", "Paint"))
        self.actionErase.setText(_translate("View", "Erase"))
        self.actionThick.setText(_translate("View", "Thick"))
        self.actionMedium.setText(_translate("View", "Medium"))
        self.actionThin.setText(_translate("View", "Thin"))
        self.actionRed.setText(_translate("View", "Red"))
        self.actionBlue.setText(_translate("View", "Blue"))
        self.actionYellow.setText(_translate("View", "Yellow"))
        self.actionUndo.setText(_translate("View", "Undo"))
        self.actionRedo.setText(_translate("View", "Redo"))
        self.actionClear_all.setText(_translate("View", "Clear all"))
        self.actionOpen.setText(_translate("View", "Open"))
        self.actionSave.setText(_translate("View", "Save"))
        self.actionCalibrate.setText(_translate("View", "Calibrate"))
        self.actionHelp.setText(_translate("View", "Help"))
        self.actionColorpicker.setText(_translate("View", "Colorpicker"))
        self.actionStrokewidth.setText(_translate("View", "Strokewidth"))

